ARM STRONG NUMBERS
a number is set to be arm strong if and only if the original number is same as the sum of exponent value (where the base is the individual digit and power is count of digits).  

constraints :
-10000000 <= n <= 1000000


Arm strong 
1. wap in dispaly the asn number with customized function
2. wap to display all the asn's present in a user defined range 
3. wap to display all the asn's present in a user defined range separalety
4. wap to display first n asn's 
5. wap to display first n not asn's 


In [1]:
# wap in dispaly the asn number with customized function
def CountDigit(n):
    count=0
    while n>0:
        n=n//10
        count+=1
    return count

def isASN(n):
    temp=n
    if n<0:
        n=n*-1

    asn, pow=0, CountDigit(n)

    while n>0:
        base =n%10
        asn=asn+(base**pow)
        n=n//10

    if temp<0:
        asn=asn*-1

    return temp==asn


num=int(input("Enter the number :"))
if isASN(num):
    print(num,"is a ASN")
else:
    print(num,"is not a ASN")

153 is a ASN


### Question 2
Write a Python program for: **wap to display all the asn's present in a user defined range**

In [ ]:
# wap to display all the asn's present in a user defined range 


def CountDigit(n):
    count=0
    while n>0:
        n=n//10
        count+=1
    return count

def isASN(n):
    temp=n
    if n<0:
        n=n*-1

    asn, pow=0, CountDigit(n)

    while n>0:
        base =n%10
        asn=asn+(base**pow)
        n=n//10

    if temp<0:
        asn=asn*-1

    return temp==asn

sn=int(input("enter the start number "))
en=int(input("enter the end number "))

if sn>en:
    print("invalid number")
else:
    print("asn numnber :" )
    for i in range(sn, en+1):
        if isASN(i):
            print(i, end=" ")

        



asn numnber :
153 

### Question 3
Write a Python program for: **wap to display all the asn's present in a user defined range separalety**

In [3]:

# wap to display all the asn's present in a user defined range separalety

def CountDigit(n):
    count=0
    while n>0:
        n=n//10
        count+=1
    return count

def isASN(n):
    temp=n
    if n<0:
        n=n*-1
    asn, pow=0, CountDigit(n)

    while n>0:
        base =n%10
        asn=asn+(base**pow)
        n=n//10

    if temp<0:
        asn=asn*-1

    return temp==asn

sn=int(input("enter the start number "))
en=int(input("enter the end number "))

if sn>en:
    print("invalid number")
else:
    print("asn numnber :" )
    for i in range(sn, en+1):
        if isASN(i):
            print(i, end=" ")


    print("\n")

    print("not a asn : ")
    for i in range (sn, en+1):
        if not isASN(i):
            print(i,end=" ")


asn numnber :
153 

not a asn : 
100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 

### Question 4
Write a Python program for: **wap to display first n asn's**

In [ ]:
# wap to display first n asn's 

def CountDigit(n):
    count=0
    while n>0:
        n=n//10
        count+=1
    return count

def isASN(n):
    temp=n
    if n<0:
        n=n*-1
    asn, pow=0, CountDigit(n)

    while n>0:
        base =n%10
        asn=asn+(base**pow)
        n=n//10
  
    if temp<0:
        asn=asn*-1

    return temp==asn
  


count = int(input("Enter how many Armstrong numbers: "))

series = 1

while count > 0:
    if isAsn(series):
        print(series, end=" ")
        count -= 1
    series += 1

1 2 3 4 5 6 7 8 9 153 370 371 407 1634 8208 9474 54748 92727 93084 548834 

### Question 5
Write a Python program for: **wap to dispaly first n non asn asn's**

In [12]:
# wap to dispaly first n non asn asn's 


def CountDigit(n):
    count=0
    while n>0:
        n=n//10
        count+=1
    return count

def isASN(n):
    temp=n
    if n<0:
        n=n*-1
    asn, pow=0, CountDigit(n)

    while n>0:
        base =n%10
        asn=asn+(base**pow)
        n=n//10

    if temp<0:
        asn=asn*-1

    return temp==asn
  


count=int(input("enter the number "))
series=1
while count>0:
    if not isAsn(series):
        print(series, end=" ")
        count-=1
    series+=1

10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 

Reverse a number
1. write a code to revese to  a number using customize function
2. wap to reverse a number present in a user defined range
3. wap to display the first n natural nume in revse order 

eg: n=15
1 2 3 4 5 6 7 7 9 1 11 21 31 41 51 

In [25]:
# write a code to revese to  a number using customize function

def integerReversal(n):
    temp=n
    if n<0:
         n=n*-1
    rev=0

    while n>0:
        rem =n%10
        rev=(rev*10)+rem
        n=n//10

    if temp<0:
        rev=rev*-1

    return rev 

num=int(input("Enter the number :"))
res = integerReversal(num)
print("the reversal of ", num,"is :", res)



the reversal of  1563 is : 3651


### Question 7
Write a Python program for: **wap to reverse a number present in a user defined range**

In [29]:
# wap to reverse a number present in a user defined range
def integerReversal(n):
    temp=n
    if n<0:
        n=n*-1
    rev=0
    
    while n>0:
        rem=n%10
        rev=(rev*10)+rem
        n=n//10

    if temp<0:
        rev=rev*-1
    
    return rev


sr=int(input("enter the  starting range : "))
er= int(input("enter the ending range: "))

if sr>er:
    print("invalid numebr")

else:
    print(" the reverse number : ")
    for i in range (sr, er+1):
        res = integerReversal(i)
        print(res ,end=" ")

 the reverse number : 
11 21 31 41 51 61 71 81 91 2 

### Question 8
Write a Python program for: **wap to display the first n natural num in reve order**

In [ ]:
# wap to display the first n natural num in reve order 

# eg: n=15
# 1 2 3 4 5 6 7 7 9 1 11 21 31 41 51 

def reverNum(n):
    temp=n
    if n<0:
        n=n*-1
    rev=0

    while n>0:
        rem=n%10
        rev=(rev*10)+rem
        n=n//10

    if temp <0:
        rev=rev*-1

    return rev

count=int(input("enter the number : "))
series=1

while count>0:
    print(reverNum(series), end=" ")
    count -= 1
    series += 1


1 2 3 4 5 6 7 8 9 1 

integer palindrome 
1. cheak weather the number is palindrome or not 
2. cheak weather the number is palindrome in the user range 
3. wap of the first plaindrome number 
4. wap of the first not palindrome number 


In [17]:
# 1. cheak weather the number is palindrome or not 
def num_Palindrome(n):
    temp=n
    if n<0:
        n=n*-1
    rev=0
    
    while n>0:
        rem=n%10
        rev=(rev*10)+rem
        n=n//10

    if temp<0:
        rev=rev*-1

    return rev==temp


num=int(input("enter the numbr : "))
if num_Palindrome(num):
    print("the palindrome number is",num)

else:
    print("not a palindrome number ", num )
  

not a palindrome number  65898


### Question 10
Write a Python program for: **2. cheak weather the number is palindrome in the user range**

In [22]:
# 2. cheak weather the number is palindrome in the user range

def num_Palindrome(n):
    temp=n
    if n<0:
        n=n*-1
    rev=0

    while n>0:
        rem=n%10
        rev=(rev*10)+rem  
        n=n//10
    
    if temp <0:
        rev=rev*-1
    

    return rev==temp

sr=int(input("enter the starting number : "))
er = int(input("enter the end number : "))

if sr >er:
    print("invalided number ")

else:
    print("the first palindrome number : ")
    for i in range (sr, er+1):
       if num_Palindrome(i):
        print(i,end=" ")

    



the first palindrome number : 
101 111 121 131 141 151 161 171 181 191 202 212 222 232 242 252 262 272 282 292 

### Question 11
Write a Python program for: **3.wap of the first plaindrome number**

In [25]:
# 3.wap of the first plaindrome number 

def num_Palindrome(n):
    temp=n
    if n<0:
        n=n*-1
    rev=0

    while n>0:
        rem=n%10
        rev=(rev*10)+rem  
        n=n//10
    
    if temp <0:
        rev=rev*-1
    

    return rev==temp

count=int(input("enter the number: "))
series=1

while count>0:
    if num_Palindrome(series):
        print(series,end=" ")
        count-=1

    series+=1

1 2 3 4 5 6 7 8 9 11 22 33 44 55 66 77 88 99 101 111 

### Question 12
Write a Python program for: **4.wap of the first  not a plaindrome number**

In [26]:
 # 4.wap of the first  not a plaindrome number 

def num_Palindrome(n):
    temp=n
    if n<0:
        n=n*-1
    rev=0

    while n>0:
        rem=n%10
        rev=(rev*10)+rem  
        n=n//10
    
    if temp <0:
        rev=rev*-1
    

    return rev==temp

count=int(input("enter the number: "))
series=1

while count>0:
    if not num_Palindrome(series):
        print(series,end=" ")
        count-=1

    series+=1



10 12 13 14 15 16 17 18 19 20 21 23 24 25 26 27 28 29 30 31 

### Question 13
Write a Python program for: **Python Program Implementation 13**

In [29]:
def rev(n):
    temp=n
    a=0
    if n<0:
        n*=-1
    
    while(n>0):
        b=n%10
        a=(a*10)+b
        n//=10
    if temp<0:
        a*=-1
    return a
n=int(input())
q=rev(n)
print(q)

4321


## Theory & Key Concepts Used in the Codes

1. **Armstrong Number**: A number equal to the sum of its digits raised to the power of total digits (e.g. $153 = 1^3 + 5^3 + 3^3$).
2. **Power Operator (`**`)**: `digit ** count` computes power of digit.
3. **Digit Extraction & Reduction**: `digit = n % 10` gets digit, `n = n // 10` drops digit.
4. **Number Reversal**: `rev = rev * 10 + digit` constructs reverse integer.
5. **Palindrome Number**: Number equal to its reverse (`num == rev`).